vividus-engine/src/main/java/org/vividus/batch/BatchStorage.java

```diff
@@ -24,23 +24,38 @@
 
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.vividus.util.property.IPropertyMapper;
 
 public class BatchStorage
 {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BatchStorage.class);
+
     private static final String BATCH = "batch-";
 
     private final Map<String, BatchConfiguration> batchConfigurations;
-
     private final Duration defaultStoryExecutionTimeout;
     private final List<String> defaultMetaFilters;
     private final boolean failFast;
 
-    public BatchStorage(IPropertyMapper propertyMapper, String defaultStoryExecutionTimeout,
-            List<String> defaultMetaFilters, boolean failFast) throws IOException
+    public BatchStorage(IPropertyMapper propertyMapper, Duration defaultStoryExecutionTimeout,
+        String deprecatedDefaultStoryExecutionTimeout, List<String> defaultMetaFilters,
+            boolean failFast) throws IOException
     {
         this.defaultMetaFilters = defaultMetaFilters;
-        this.defaultStoryExecutionTimeout = Duration.ofSeconds(Long.parseLong(defaultStoryExecutionTimeout));
+        if (null != deprecatedDefaultStoryExecutionTimeout)
+        {
+            this.defaultStoryExecutionTimeout =
```

Comment:
I think we need to fail if both deprecated and non-deprecated properties are specified